ckanext/cfpb_extrafields/templates/package/snippets/resource_type-specific/cfpb_extrafields_database.html

```diff
@@ -47,24 +47,19 @@
   <div>
     <p>Roles and access descriptions:</p>
   </div>
-  {{ form.input('db_role_level_1', type='text', label=_("Level 1"),
-  placeholder="", value=data.db_role_level_1,error=errors.db_role_level_1,classes=['control-full']) }}
-  {{ form.input('db_role_level_2', type='text', label=_("Level 2"),
-  placeholder="", value=data.db_role_level_2,error=errors.db_role_level_2,classes=['control-full']) }}
-  {{ form.input('db_role_level_3', type='text', label=_("Level 3"),
-  placeholder="", value=data.db_role_level_3,error=errors.db_role_level_3,classes=['control-full']) }}
-  {{ form.input('db_role_level_4', type='text', label=_("Level 4"),
-  placeholder="", value=data.db_role_level_4,error=errors.db_role_level_4,classes=['control-full']) }}
-  {{ form.input('db_role_level_5', type='text', label=_("Level 5"),
-  placeholder="", value=data.db_role_level_5,error=errors.db_role_level_5,classes=['control-full']) }}
-  {{ form.input('db_role_level_6', type='text', label=_("Level 6"),
-  placeholder="", value=data.db_role_level_6,error=errors.db_role_level_6,classes=['control-full']) }}
-  {{ form.input('db_role_level_7', type='text', label=_("Level 7"),
-  placeholder="", value=data.db_role_level_7,error=errors.db_role_level_7,classes=['control-full']) }}
-  {{ form.input('db_role_level_8', type='text', label=_("Level 8"),
-  placeholder="", value=data.db_role_level_8,error=errors.db_role_level_8,classes=['control-full']) }}
-  {{ form.input('db_role_level_9', type='text', label=_("Level 9"),
-  placeholder="", value=data.db_role_level_9,error=errors.db_role_level_9,classes=['control-full']) }}
+
+  {# Add 20 slots for blank rows to be added #}
+  {% set roles = data.get("db_roles", []) + [("", "")] * 20 %}
```

Comment:
Basically the code creates 20 slots, but they are all hidden in the webpage. When you click "add role", we make the first hidden set of inputs visible. Therefore, having a big number here doesn't really have any downsides besides adding a few bytes to the page size.

Perhaps I should push this out to the staging instance so that you can take a look there and verify that you are happy with how this looks before we consider merging.